internal/service/dns_config/api_auth_zone_data_source.go

```diff
@@ -98,18 +100,30 @@ func (d *AuthZoneDataSource) Read(ctx context.Context, req datasource.ReadReques
 		return
 	}
 
-	apiRes, _, err := d.client.DNSConfigurationAPI.
-		AuthZoneAPI.
-		AuthZoneList(ctx).
-		Filter(flex.ExpandFrameworkMapFilterString(ctx, data.Filters, &resp.Diagnostics)).
-		Tfilter(flex.ExpandFrameworkMapFilterString(ctx, data.TagFilters, &resp.Diagnostics)).
-		Execute()
-	if err != nil {
-		resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read AuthZone, got error: %s", err))
-		return
+	for {
+		apiRes, _, err := d.client.DNSConfigurationAPI.
+			AuthZoneAPI.
+			AuthZoneList(ctx).
+			Filter(flex.ExpandFrameworkMapFilterString(ctx, data.Filters, &resp.Diagnostics)).
+			Tfilter(flex.ExpandFrameworkMapFilterString(ctx, data.TagFilters, &resp.Diagnostics)).
+			Offset(offset).
+			Limit(limit).
+			Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read AuthZone, got error: %s", err))
+			return
+		}
+
+		allResults = append(allResults, apiRes.GetResults()...)
+
+		if len(apiRes.GetResults()) < int(limit) {
+			break
+		}
+
+		offset += limit
```

Comment:
This can possibly be refactored and moved to a common function?

Reply:
fixed